### DISCUSSION TITLE
Trigger on inserts causes 500 error on signup

### DISCUSSION THREAD

**Discussion by kaelaworthen** [supabase/supabase#26979] [Questions] [answered]
I have a trigger set after insert on auth.users to run a function called auth_new_signup(). This function then takes the newly created user, creates a row in the profiles table and connects them, creates a row in the estates table and connects that to the profile, and creates a bunch of rows in the task_assignments table that are connected to that same estate.

When I try to add triggers on the profiles, estates, or task_assignments table that are related to inserts, however, the supabase signup flow breaks and returns a 500 error.

It does NOT break if:
* the trigger on those tables only runs on update, not insert and update
* the trigger runs on insert and update, but I remove the trigger from the auth.users table
* the trigger runs on a table other than profiles, estates, or task_assignments (aka a table not used in the auth_new_signup function)

It DOES break if:
* there is a trigger for inserts in addition to/instead of updates AND that table is also used in the auth_new_signup function
* regardless of whether the functions are security definer functions or not

Any ideas?

My actual goal is to create a notification any time there is an insert or an update on task_assignments, but I can't do it for inserts until this is resolved.

**Comment by GaryAustin1** [comment]
As commented on a previous post you need to look to the Postgres logs and find what the actual error is.  It will also usually tell you the function if it is in that.

Anything started from an auth trigger is impacted by the auth role not having permissions (security definer can solve that), not have public in the search path (set the search_path in the functions or use public.table for every reference), and then the NEW and OLD data that may or not be there depending on the the operation.   Also you could get errors trying to access data in a table that is not really inserted/updated yet if you select it in other trigger functions.

But to start find out what your error is.

**Reply by kaelaworthen** [reply]
I can't figure out the errors I'm getting or why. I have all the schemas identified and the logging isn't proving useful to me.

Here's the full function for signing up, with progressively more and more logging added:

```
redacted
```

This function and trigger does work as expected, and the signup endpoint also works as expected.

But when I add this trigger below, then for some reason the signup endpoint breaks:

```
redacted
```

You'll notice both functions have a security definer, and both have the schema defined for public vs auth. The signup only breaks when the notifications trigger is set for both insert and update; if the notif_create_notification is only on update, then everything works fine.

When signing up a new user with both functions in place as above, I get these logs:

Console:
```
redacted
```

I can't figure out what this means since the profiles table should be clearly defined in the functions.

Supabase:
```
redacted
```

These logs don't seem to have a lot of information that I can figure out how to parse and use.

**Reply by GaryAustin1** [reply, marked as answer]
In some function you have profiles without public. In front of it.  It could be in a trigger function you forgot about or above. I'm on phone and did not syntax check then.

**Reply by GaryAustin1** [reply]
The Postgres log should have more detail on the error.

**Reply by kaelaworthen** [reply]
Ah, I didn't realize I needed every single trigger to have public. in front of the tables. That fixed it, thanks!

**Reply by GaryAustin1** [reply]
If they are initiated with an auth call versus a database API call.
